examples/regression/4-tutorials/plot_main-tutorial-regression.py

```diff
@@ -0,0 +1,795 @@
+# function of degree 10 and show the results for the Jackknife+ and CV+
+# strategies.
+
+polyn_model_quant = Pipeline(
+    [
+        ("poly", PolynomialFeatures(degree=degree_polyn)),
+        ("linear", QuantileRegressor(
+                solver="highs-ds",
+                alpha=0,
+        ))
+    ]
+)
+STRATEGIES = {
+    "naive": dict(method="naive"),
+    "jackknife": dict(method="base", cv=-1),
+    "jackknife_plus": dict(method="plus", cv=-1),
+    "jackknife_minmax": dict(method="minmax", cv=-1),
+    "cv": dict(method="base", cv=10),
+    "cv_plus": dict(method="plus", cv=10),
+    "cv_minmax": dict(method="minmax", cv=10),
+    "jackknife_plus_ab": dict(method="plus", cv=Subsample(n_resamplings=50)),
+    "jackknife_minmax_ab": dict(
+        method="minmax", cv=Subsample(n_resamplings=50)
+    ),
+    "conformalized_quantile_regression": dict(
+        method="quantile", cv="split", alpha=0.05
+    )
+}
+y_pred, y_pis = {}, {}
+for strategy, params in STRATEGIES.items():
+    if strategy == "conformalized_quantile_regression":
+        mapie = MapieQuantileRegressor(polyn_model_quant, **params)
+        mapie.fit(X_train, y_train, random_state=1)
+        y_pred[strategy], y_pis[strategy] = mapie.predict(X_test)
+    else:
+        mapie = MapieRegressor(polyn_model, **params)
+        mapie.fit(X_train, y_train)
+        y_pred[strategy], y_pis[strategy] = mapie.predict(X_test, alpha=0.05)
+
+strategies = [
+    "jackknife_plus",
+    "jackknife_minmax",
+    "cv_plus",
+    "cv_minmax",
+    "jackknife_plus_ab",
+    "conformalized_quantile_regression"
+]
+n_figs = len(strategies)
+fig, axs = plt.subplots(3, 2, figsize=(9, 13))
+coords = [axs[0, 0], axs[0, 1], axs[1, 0], axs[1, 1], axs[2, 0], axs[2, 1]]
+for strategy, coord in zip(strategies, coords):
+    plot_1d_data(
+        X_train.ravel(),
+        y_train.ravel(),
+        X_test.ravel(),
+        y_mesh.ravel(),
+        1.96*noise,
+        y_pred[strategy].ravel(),
+        y_pis[strategy][:, 0, :].ravel(),
+        y_pis[strategy][:, 1, :].ravel(),
+        ax=coord,
+        title=strategy
+    )
+
+
+##############################################################################
+# At first glance, our polynomial function does not give accurate
+# predictions with respect to the true function when :math:`|x > 6|`.
+# The prediction intervals estimated with the Jackknife+ do not seem to
+# increase significantly, unlike the CV+ method whose prediction intervals
+# capture a high uncertainty when :math:`x > 6`.
+#
+# Let's now compare the prediction interval widths between all strategies.
+
+fig, ax = plt.subplots(1, 1, figsize=(7, 5))
+ax.set_yscale("log")
+for strategy in STRATEGIES:
+    ax.plot(
+        X_test,
+        y_pis[strategy][:, 1, 0] - y_pis[strategy][:, 0, 0],
+        label=strategy
+    )
+ax.set_xlabel("x")
+ax.set_ylabel("Prediction Interval Width")
+ax.legend(fontsize=10, loc=[1, 0.4])
```

Comment:
add a plt.show() to remove the log from matplotlib

Reply:
Done